addons/mail/static/src/models/file_uploader_view/file_uploader_view.js

```diff
@@ -4,10 +4,147 @@ import { registerModel } from '@mail/model/model_core';
 import { attr, many2one, one2one } from '@mail/model/model_field';
 import { clear, replace } from '@mail/model/model_field_command';
 
+import core from 'web.core';
+
+const geAttachmentNextTemporaryId = (function() {
+    let tmpId = 0;
+    return () => {
+        tmpId += 1;
+        return tmpId;
+    };
+})();
+
 registerModel({
     name: 'FileUploaderView',
     identifyingFields: [['activityView', 'attachmentBoxView', 'composerView']],
+    lifecycleHooks: {
+        _created() {
+            this.onChangeAttachment = this.onChangeAttachment.bind(this);
+        }
+    },
     recordMethods: {
+        /**
+         * @param {FileList|Array} files
+         * @returns {Promise}
+         */
+        async uploadFiles(files) {
+            await this._performUpload({ files });
+            this.fileInputRef.el.value = '';
+        },
+        openBrowserFileUploader() {
+            this.fileInputRef.el.click();
```

Comment:
ideally needs to be guarded too, but it is not very critical (it could be possible by being very unlucky to click on the add button when child component is already destroyed)